source/codegen/metadata_validation.py

```diff
@@ -86,6 +98,11 @@ class RULES:
         Optional("additional_arguments_to_output_allocation"): [str],
         Optional("proto_only"): bool,
         Optional("input_passed_by_ptr"): bool,
+        Optional("cppName"): str,
+        Optional("determine_size_from"): [str],
+        Optional("is_size_param"): bool,
+        Optional("linked_params_are_optional"): bool,
+        Optional("mapped-enum"): str,
```

Comment:
This entry `mapped-enum` is the only one in the list I see with the word separator being a dash instead of an underscore. Is that intentional? Can we fix it in the tool?

While we're at it we should probably also fix `cppName` which is the only one that doesn't have a word separator?

Reply:
Hmm, I don't think there was any intention behind it. Nor with `cppName`. I just don't think we standardized those with the rest.

Also, there's a handful of other dash-separated keys and values throughout the metadata files. We seemed to have added some for Enums along the way, like 'generate-mappings' and 'force-include'.

Then there were some pre-existing ones (from nimi-python and in turn hapigen) for the parameter's size mechanisms like 'ivi-dance-with-a-twist' and 'passed-in'.

Reply:
We can either

a) Hold off on them and standardize on underscore_seperated keys/values in a future clean-up.

or

b) I could address mapped-enum and cppName right now. Replace all instances with underscore separated versions. I would also then need to make corresponding updates to [metadata_mutation.py](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/hapigen/hapigen_grpc_device_plugin/source/hapigen_grpc_device_plugin/metadata_mutation.py&version=GBmain&line=173&lineEnd=174&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) used for the hapigen plugin.

Reply:
I'd honestly prefer tech debt, at some theoretical point when we're not being quizzed every day as to progress.

Reply:
Sounds good. Made AB#2146456 in our Tech Debt backlog to look address this.